scripts/apps/authoring/authoring/directives/ReadingTime.js

```diff
@@ -0,0 +1,34 @@
+import * as helpers from 'apps/authoring/authoring/helpers';
+
+ReadingTime.$inject = ['gettextCatalog'];
+export function ReadingTime(gettextCatalog) {
+    return {
+        scope: {
+            item: '=',
+            html: '@'
+        },
+        template: '<span class="char-count reading-time">' +
+            '<span translate> ' + gettextCatalog.getString('{{readingTime}} min read</span></span>'),
+        link: function ReadingTimeLink(scope, elem, attrs) {
+            scope.html = scope.html || false;
+            scope.$watch('item', () => {
+                var input = scope.item || '';
```

Comment:
You are declaring the variables here and assigning them values immediately below. Seems a bit repetitive. You can just do:
```js
let {html, item} = scope;
let input = html ? helpers.cleanHtml(item || '') : item || '';
let numWords = _.compact(input.split(/\s+/)).length || 0;
let readingTimeFloat = numWords / 250;
let readingTimeMinutes = Math.floor(readingTimeFloat);
let readingRemainingSec = Math.floor((readingTimeFloat - readingTimeMinutes) * 60);

if (readingRemainingSec >= 30) {
   readingTimeMinutes++;
}
```
Seems more succinct to me.